fossunited/www/book-conference-ticket/index.py

```diff
@@ -0,0 +1,189 @@
+            "pin_code": values.get("pin_code"),
+            "email": values.get("email"),
+            "phone": values.get("phone"),
+            "state": values.get("state"),
+            "general_tickets": values.get("general_tickets"),
+            "student_tickets": values.get("student_tickets"),
+        }
+    )
+
+    total_amount = (
+        float(student_ticket_price) * int(ticket.student_tickets)
+    ) + (float(general_ticket_price) * int(ticket.general_tickets))
+
+    ticket.total_amount = total_amount
+
+    ticket.insert()
+    ticket.order_id = makeOrderId(ticket.name, total_amount)
+    ticket.save()
+
+    return {
+        "ticket_id": ticket.name,
+        "order_id": ticket.order_id,
+        "amount": ticket.total_amount,
+        "razorpay_key": frappe.db.get_single_value(
+            "Razorpay Keys", "rzp_key"
+        ),
+        "currency": "INR",
+        "fullname": ticket.first_name + " " + ticket.last_name,
+        "email": ticket.email,
+    }
+
+
+@frappe.whitelist()
+def makeOrderId(ticket_id, total_amount):
+    rzp_basic_auth = f"Basic {createBase()}"
+
+    order = make_post_request(
+        f"https://api.razorpay.com/v1/orders",
+        headers={
+            "Authorization": rzp_basic_auth,
+            "Content-Type": "application/json",
+        },
+        data=json.dumps(
+            {
+                "amount": total_amount * 100,
+                "currency": "INR",
+                "notes": {"ticket_id": ticket_id},
+            }
+        ),
+    )
+
+    return order["id"]
+
+
+@frappe.whitelist()
+def getPaymentId(order_id):
+    rzp_basic_auth = f"Basic {createBase()}"
+
+    payment = make_get_request(
+        f"https://api.razorpay.com/v1/orders/{order_id}/payments",
+        headers={
+            "Authorization": rzp_basic_auth,
+            "Content-Type": "application/json",
+        },
+    )
+
+    return payment["items"][0]["id"]
+
+
+# Function to auomate the formatting of api_key:api_secret
+def createBase():
+    key_id = frappe.db.get_single_value("Razorpay Keys", "rzp_key")
+    secretKey = frappe.get_single("Razorpay Keys")
+    key_secret = secretKey.get_password("key_secret")
+    basic_auth = base64.b64encode(
+        f"{key_id}:{key_secret}".encode()
+    ).decode()
+
+    return basic_auth
+
+
+# Function to update captured payment state
+@frappe.whitelist()
+def capture_payment(doctype, ticket_id):
+    doc = frappe.get_doc("Conference Tickets", ticket_id)
+
+    basic_auth = f"Basic {createBase()}"
+
+    if not doc.payment_captured:
+        doc.razorpay_payment_id = getPaymentId(doc.order_id)
+
+        order = make_get_request(
+            f"https://api.razorpay.com/v1/orders/{doc.order_id}",
+            headers={
+                "Authorization": basic_auth,
+                "Content-Type": "application/json",
+            },
+        )
+
+        doc.payment_captured = (
+            True if order["status"] == "paid" else False
```

Comment:
order status will be paid or captured?

Reply:
it will be paid, the razorpay documentation marks order as paid too. 
![image](https://github.com/fossunited/fossunited/assets/82205152/9bd603f7-ada9-44be-a8b6-f16331512516)